src/lib/optimizer/strategy/join_predicate_ordering_rule.cpp

```diff
@@ -0,0 +1,69 @@
+#include "join_predicate_ordering_rule.hpp"
+
+#include "cost_estimation/abstract_cost_estimator.hpp"
+#include "expression/abstract_predicate_expression.hpp"
+#include "expression/expression_utils.hpp"
+#include "expression/logical_expression.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "statistics/cardinality_estimator.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+void JoinPredicateOrderingRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) const {
+  // Check if this is a multi predicate join.
```

Comment:
Please use visit_lqp when traversing an LQP as it avoids traversing subplans below diamonds redundantly